packages/itwin/grouping-mapping-widget/src/widget/components/Grouping.tsx

```diff
@@ -237,11 +237,20 @@ export const Groupings = ({
     await zoomToElements(allIds);
   }, [setHiddenGroupsIds, groups, hideGroupsWrapper, getHiliteIdsFromGroupsWrapper]);
 
+  const ToggleGroupVis = () => <ToggleSwitch
+    label='Color by Group'
+    labelPosition='left'
+    className='gmw-toggle'
+    disabled={isLoadingQuery}
+    checked={showGroupColor}
+    onChange={() => setShowGroupColor((b) => !b)}
+  ></ToggleSwitch>;
+
   return (
     <>
       <Surface className='gmw-groups-container'>
         <div className='gmw-toolbar'>
-          {onClickAddGroup && groupUIs.length > 0 &&
+          {onClickAddGroup && groupUIs.length > 0 ?
```

Comment:
It's pretty hard to understand where different parts of this ternary operator start/end. Maybe moving the whole `DropdownMenu` into a separate component would help.

Reply:
Extracted the ```DropDownmenu```/```Button``` to its own file.